pkg/reconciler/inmemorychannel/dispatcher/inmemorychannel.go

```diff
@@ -50,15 +51,36 @@ type Reconciler struct {
 	messagingClientSet         messagingv1.MessagingV1Interface
 }
 
+// Check the interfaces Reconciler should implement
+var (
+	_ reconcilerv1.Interface         = (*Reconciler)(nil)
+	_ reconcilerv1.ReadOnlyInterface = (*Reconciler)(nil)
+)
+
+// ReconcileKind implements inmemorychannel.Interface.
 func (r *Reconciler) ReconcileKind(ctx context.Context, imc *v1.InMemoryChannel) reconciler.Event {
+	if err := r.reconcile(ctx, imc); err != nil {
+		return err
+	}
```

Comment:
This is pretty much all this PR does: move the reconciliation logic to a `(*Reconciler).reconcile()` method and call it from both `ReconcileKind()` and `ObserveKind()`.